pkgs/development/perl-modules/generic/builder.sh

```diff
@@ -4,38 +4,39 @@ PERL5LIB="$PERL5LIB${PERL5LIB:+:}$out/lib/perl5/site_perl"
 
 perlFlags=
 for i in $(IFS=:; echo $PERL5LIB); do
-    perlFlags="$perlFlags -I$i"
+    if [[ $i == "$hostperl/lib/perl5/site_perl" ]] || [[ $i == "$hostperl/lib/perl5/site_perl/"* ]]; then
+        # exclude paths pointing to perl derivation (like /nix/store/abcdfghijklmnpqrsvwxyz0123456789-perl-5.28.2-aarch64/lib/perl5/site_perl/5.28.2)
+        # they are needless when pointing to correct perl, and source of subtle bugs when pointing to wrong perl
+        true
+    elif [[ $i =~ /[0123456789abcdfghijklmnpqrsvwxyz]{32}-perl-[0-9.]+(-[^/]+)?/lib/perl5/site_perl($|/) ]]; then
+        echo -e "unexpected perl $i on PERL5LIB where\n hostperl=$hostperl\n devperl =$devperl\n fullperl=$fullperl"
```

Comment:
use `disallowedReferences` in the builder instead?

Reply:
Hm, no.
1. It is difficult to build the list for `disallowedReferences`.
I am worrying of a case of leaking `pkgs.perl` (or `pkgs.perl.withPackages ....`, or `pkgsCross.*.perl.withPackages ....`) into packages targeting non-default perl (`perldevel` or `perl530`) more than about closure size. There has been such a bug few months ago.
The regex covers all the variety of `(pkgsCross.*|pkgsi686|)perl*(.withPackages|)` packages except `$hostperl` which is checked in the `if` above.
2. This assertion will trigger even if the wrong derivation path does not land to `$out`, it is a signal that `buildPerlPackage`'s contract is misused.